packages/api/server.js

```diff
@@ -56,6 +71,65 @@ app.get('/api/repos', async (req, res) => {
     .end();
 });
 
+app.get('/githubauth', (req, res) => {
+  const url = 'http://github.com/login/oauth/authorize?client_id=' + process.env.CLIENT_ID + '&state=MUSTCHANGE&allow_signup=false';
+  res.status(302).redirect(url);
+});
+
+app.get('/api/callback', (req, res) => {
+  if (req.param('state') !== 'MUSTCHANGE') {
+    res.status(401).send('Error: invalid state!');
+    return;
+  }
+
+  request.post(
+    {
+      headers: { 'content-type': 'application/json', Accept: 'application/json' },
+      url: 'https://github.com/login/oauth/access_token',
+      body: JSON.stringify({
+        code: req.param('code'),
+        client_id: process.env.CLIENT_ID,
+        client_secret: process.env.CLIENT_SECRET,
+        state: 'MUSTCHANGE' // TODO: make this something better
```

Comment:
Not blocking:
Maybe save _MUSTCHANGE_ it in a const ? (since its at line 75 and 80)

Reply:
Yes, this will likely ultimately be a uuid value.